shap/plots/_bar.py

```diff
@@ -54,17 +55,35 @@ def bar(shap_values, max_display=10, order=Explanation.abs, clustering=None, clu
         cohorts = {"": shap_values}
     elif isinstance(shap_values, Cohorts):
         cohorts = shap_values.cohorts
+    elif isinstance(shap_values, dict):
+        cohorts = shap_values
     else:
-        assert isinstance(shap_values, dict), "You must pass an Explanation object, Cohorts object, or dictionary to bar plot!"
+        emsg = (
+            "The shap_values argument must be an Explanation object, Cohorts "
+            "object, or dictionary of Explanation objects!"
+        )
+        raise TypeError(emsg)
 
     # unpack our list of Explanation objects we need to plot
     cohort_labels = list(cohorts.keys())
     cohort_exps = list(cohorts.values())
-    for i in range(len(cohort_exps)):
-        if len(cohort_exps[i].shape) == 2:
-            cohort_exps[i] = cohort_exps[i].abs.mean(0)
-        assert isinstance(cohort_exps[i], Explanation), "The shap_values paramemter must be a Explanation object, Cohorts object, or dictionary of Explanation objects!"
-        assert cohort_exps[i].shape == cohort_exps[0].shape, "When passing several Explanation objects they must all have the same shape!"
+    for i, exp in enumerate(cohort_exps):
+        if not isinstance(exp, Explanation):
+            emsg = (
+                "The shap_values argument must be an Explanation object, Cohorts "
+                "object, or dictionary of Explanation objects!"
+            )
+            raise TypeError(emsg)
```

Comment:
Note that the order of checks has switched. I check for Explanation object first, because otherwise some invalid inputs won't have the `shape`  attribute (used in the check on the next line).